src/storage/irc_table_entry.cpp

```diff
@@ -0,0 +1,131 @@
+#include "storage/irc_transaction.hpp"
+#include "duckdb/storage/statistics/base_statistics.hpp"
+#include "duckdb/storage/table_storage_info.hpp"
+#include "duckdb/main/extension_util.hpp"
+#include "duckdb/main/database.hpp"
+#include "duckdb/main/secret/secret_manager.hpp"
+#include "duckdb/catalog/catalog_entry/table_function_catalog_entry.hpp"
+#include "duckdb/parser/tableref/table_function_ref.hpp"
+#include "catalog_api.hpp"
+#include "../../duckdb/third_party/catch/catch.hpp"
+#include "duckdb/planner/binder.hpp"
+#include "duckdb/planner/tableref/bound_table_function.hpp"
+#include "duckdb/planner/logical_operator.hpp"
+#include "duckdb/planner/operator/logical_get.hpp"
+
+
+namespace duckdb {
+
+ICTableEntry::ICTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, CreateTableInfo &info)
+    : TableCatalogEntry(catalog, schema, info) {
+	this->internal = false;
+}
+
+ICTableEntry::ICTableEntry(Catalog &catalog, SchemaCatalogEntry &schema, ICTableInfo &info)
+    : TableCatalogEntry(catalog, schema, *info.create_info) {
+	this->internal = false;
+}
+
+unique_ptr<BaseStatistics> ICTableEntry::GetStatistics(ClientContext &context, column_t column_id) {
+	return nullptr;
+}
+
+void ICTableEntry::BindUpdateConstraints(Binder &binder, LogicalGet &, LogicalProjection &, LogicalUpdate &,
+                                         ClientContext &) {
+	throw NotImplementedException("BindUpdateConstraints");
+}
+
+TableFunction ICTableEntry::GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) {
+	auto &db = DatabaseInstance::GetDatabase(context);
+	auto &ic_catalog = catalog.Cast<ICRCatalog>();
+
+	auto &parquet_function_set = ExtensionUtil::GetTableFunction(db, "parquet_scan");
+	auto parquet_scan_function = parquet_function_set.functions.GetFunctionByArguments(context, {LogicalType::VARCHAR});
+
+	auto &iceberg_function_set = ExtensionUtil::GetTableFunction(db, "iceberg_scan");
+	auto iceberg_scan_function = iceberg_function_set.functions.GetFunctionByArguments(context, {LogicalType::VARCHAR});
+
+	D_ASSERT(table_data);
+
+	if (table_data->data_source_format != "ICEBERG") {
+		throw NotImplementedException("Table '%s' is of unsupported format '%s', ", table_data->name,
+		                              table_data->data_source_format);
+	}
+
+	if (table_data->storage_location.find("file://") != 0) {
+		auto &secret_manager = SecretManager::Get(context);
+		// Get Credentials from ICAPI
+		auto table_credentials = ICRAPI::GetTableCredentials(
+			ic_catalog.internal_name, table_data->schema_name, table_data->name, ic_catalog.credentials);
+		// First check if table credentials are set (possible the IC catalog does not return credentials)
+		if (!table_credentials.key_id.empty()) {
+			// Inject secret into secret manager scoped to this path
+			CreateSecretInfo info(OnCreateConflict::ERROR_ON_CONFLICT, SecretPersistType::TEMPORARY);
+			info.name = "__internal_ic_" + table_data->table_id;
+			info.type = "s3";
```

Comment:
Also, based on the comment [here](https://github.com/duckdb/duckdb-iceberg/pull/98#issuecomment-2633068303), we should see if the spec returns information on the type of the secret so we don't have to hard code this kind of information